router.go

```diff
@@ -229,14 +229,19 @@ func (router *Router) sendPendingGossip() bool {
 	return sentSomething
 }
 
-// BroadcastTopologyUpdate is invoked whenever there is a change to the mesh
-// topology, and broadcasts the new set of peers to the mesh.
-func (router *Router) broadcastTopologyUpdate(update []*Peer) {
-	names := make(peerNameSet)
-	for _, p := range update {
-		names[p.Name] = struct{}{}
+// BroadcastTopologyUpdate is invoked whenever there is one or more changes to the mesh
+// topology, and broadcasts the new set of peers to the mesh after coalescing updates.
+func (router *Router) broadcastTopologyUpdate(updates ...[]*Peer) {
+	var gossipData GossipData
+	gossipData = &topologyGossipData{peers: router.Peers, update: nil}
+	for _, update := range updates {
+		names := make(peerNameSet)
+		for _, p := range update {
+			names[p.Name] = struct{}{}
+		}
+		gossipData = gossipData.Merge(&topologyGossipData{peers: router.Peers, update: names})
```

Comment:
This seems unnecessary - all this `Merge` does is union the two maps, so we can just build up one set then send that at the end.